mindsdb/integrations/handlers/kraken_handler/kraken_handler.py

```diff
@@ -0,0 +1,105 @@
+import pandas as pd
+import requests
+from mindsdb.integrations.libs.api_handler import APIHandler
+from mindsdb.integrations.handlers.kraken_handler.kraken_tables import KrakenTradesTable
+from collections import OrderedDict
+
+from mindsdb.integrations.libs.const import HANDLER_CONNECTION_ARG_TYPE as ARG_TYPE
+
+from mindsdb.integrations.libs.response import HandlerStatusResponse as StatusResponse
+
+
+class KrakenHandler(APIHandler):
+    """A class for handling connections and interactions with the Kraken API.
+    Attributes:
+        api_key (str): API Public key
+        api_secret (str): API Private key
+    """
+    def __init__(self, name: str = None, **kwargs):
+        """Registers all API tables and prepares the handler for an API connection.
+        Args:
+            name: (str): The handler name to use
+        """
+        super().__init__(name)
+        self.api_key = None
+        self.api_secret = None
+
+        args = kwargs.get("connection_data", {})
+        if "api_key" in args:
+            self.api_key = args["api_key"]
+        if "api_secret" in args:
+            self.api_secret = args["api_secret"]
+        self.client = None
+        self.is_connected = False
+
+        trade_history = KrakenTradesTable(self)
+        self._register_table("kraken_trade_history", trade_history)
+
+    def connect(self):
+        """Creates a new Kraken API client if needed and sets it as the client to use for requests.
+        Returns newly created Kraken API client, or current client if already set.
+        """
+        self.is_connected = True
+        return self.client
+
+    def check_connection(self) -> StatusResponse:
+        """Checks connection to Kraken API by sending a ping request.
+        Returns StatusResponse indicating whether or not the handler is connected.
+        """
+        response = StatusResponse(True)
+        self.is_connected = response.success
+        return response
+
+    def get_trade_history(self, queries: dict) -> pd.DataFrame:
+        """Gets trade history for a user based on given parameters
+        Returns results as a pandas DataFrame.
+        Args:
+            params (Dict): Trade data params (symbol, interval)
+        """
+        api_url = "https://api.kraken.com"
+        uri_path = "/0/private/TradesHistory"
+        headers = {}
+        headers["API-Key"] = self.api_key
+        headers["API-Sign"] = self.api_secret
+        res = requests.post((api_url + uri_path), headers=headers, data=queries)
+        result = res["result"]
+        # error = res["error"]
+        trades = result["trades"]
+        trades_list = []
+        for trade_id, details in trades:
+            temp = details
+            temp["trade_id"] = trade_id
+            trades_list.append(temp)
+        df = pd.DataFrame(trades_list)
+        return df
+
+    def call_kraken_api(self, method_name, params):
```

Comment:
I am personally not the biggest fan of passing in the `method_name` as a parameter and making a call to that method like this. I believe that `get_trade_history()` function should be part of the `KrakenTradesTable` (because that is where the associated logic is) and it should be called there directly, within `select()`.
To call this method from the `APITable` implementation by passing in the method name and then have it call that particular method seems a little redundant to me.
This is not mandatory change though as it will still work.